### PR TITLE
Made some variable names more consistent with the other parts.

### PR DESCRIPTION
**Pull request by euske**
Hello, we're developing an automated system that detects inconsistent variable names in a large software project. Our system checks if each variable name is consistent with other variables in the project in its usage pattern, and proposes correct candidates if inconsistency is detected. This is a part of academic research that we hope to publish soon, but as a part of the evaluation, we applied our systems to your projects and got a few interesting results. We carefully reviewed our system output and manually created a patch to correct a few variable names. We would be delighted if this patch is found to be useful. If you have a question or suggestion regarding this patch, we'd happily
answer. Thank you.

### CODE REVIEW

File: src/main/java/org/apache/commons/compress/compressors/z/ZCompressorInputStream.java [hunk 1]
```diff
@@ -103,12 +103,12 @@ private void reAlignReading() throws IOException {
...
     protected int addEntry(final int previousCode, final byte character) throws IOException {
         final int maxTableSize = 1 << getCodeSize();
-        final int r = addEntry(previousCode, character, maxTableSize);
+        final int idx = addEntry(previousCode, character, maxTableSize);
         if (getTableSize() == maxTableSize && getCodeSize() < maxCodeSize) {
             reAlignReading();
             incrementCodeSize();
         }
-        return r;
+        return idx;
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/compressors/z/ZCompressorInputStream.java]: I am -1 to abbreviated names like "idx". I prefer "index" or, as a fallback for the brevity fans, the classic "i".

File: src/main/java/org/apache/commons/compress/utils/ByteUtils.java [hunk 2]
```diff
@@ -71,16 +71,16 @@ public static long fromLittleEndian(byte[] bytes) {
     /**
      * Reads the given byte array as a little endian long.
      * @param bytes the byte array to convert
-     * @param off the offset into the array that starts the value
+     * @param offset the offset into the array that starts the value
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/utils/ByteUtils.java]: +1

File: src/main/java/org/apache/commons/compress/archivers/zip/StreamCompressor.java [hunk 3]
```diff
@@ -259,9 +259,9 @@ private void deflateUntilInputIsNeeded() throws IOException {
     }
 
     void deflate() throws IOException {
-        final int len = def.deflate(outputBuffer, 0, outputBuffer.length);
-        if (len > 0) {
-            writeCounted(outputBuffer, 0, len);
+        final int length = def.deflate(outputBuffer, 0, outputBuffer.length);
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/archivers/zip/StreamCompressor.java]: +1

File: src/main/java/org/apache/commons/compress/archivers/zip/BitStream.java [hunk 4]
```diff
@@ -48,14 +48,14 @@ int nextBit() throws IOException {
...
      *
-     * @param n the number of bits read (up to 8)
-     * @return The value formed by the n bits, or -1 if the end of the stream has been reached
+     * @param count the number of bits read (up to 8)
+     * @return The value formed by the count bits, or -1 if the end of the stream has been reached
      */
-    long nextBits(final int n) throws IOException {
-        if (n < 0 || n > 8) {
-            throw new IOException("Trying to read " + n + " bits, at most 8 are allowed");
+    long nextBits(final int count) throws IOException {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/compress/archivers/zip/BitStream.java]: +1

**Comment by bodewig** [comment]
I am on the fence with this PR.

Personally I don't see any benefit in consistent naming of local variables as long as the local variables have names that can be understood. To me a local variable named `length` is in no way clearer than one called `len`.

We will never get a consistency at the level your tool is striving for on a project that relies on numerous volunteer contributors. I don't see myself frustrating contributors by asking them to rename their local variables on patches.

@garydgregory if you see value in this and want to merge the PR after the changes you've requested have been made I won't stand in the way but I'll otherwise ignore it myself.

**Comment by garydgregory** [comment]
I don't care that the changes come from a tool, I am just looking to see if the code is easier to read, which admittedly is subjective of course. But yeah, I can bring in the PR at some point. But if this is just a PR where the poster does not come back and update the code or comment, then it will go nowhere.

File: src/main/java/org/apache/commons/compress/compressors/z/ZCompressorInputStream.java (hunk 1 above)
Comment (euske) [reply src/main/java/org/apache/commons/compress/compressors/z/ZCompressorInputStream.java]: This suggestion for "idx" comes from a similar expression in UnshrinkingInputStream.java:

    60:        final int idx = addEntry(previousCode, character, MAX_TABLE_SIZE);
    61:        if (idx >= 0) {
    62:            isUsed[idx] = true;
    63:        }
    64:        return idx;
 
The use of "idx" is slightly more common than "index" throughout the project. In this case, there's already a variable that has a similar use, so the system picked it up. Our system doesn't create a new name but it tries to choose the best looking candidate to make the code consistent with a minimal amount of change.

**Comment by euske** [comment]
> Personally I don't see any benefit in consistent naming of local variables as long as the local variables have names that can be understood. To me a local variable named `length` is in no way clearer than one called `len`.
> 
> We will never get a consistency at the level your tool is striving for on a project that relies on numerous volunteer contributors. I don't see myself frustrating contributors by asking them to rename their local variables on patches.

I think that's a valid concern. Our system proposed numerous small changes, but I only sent some of the most conspicuous ones. I would say that overall the variable naming in this project is pretty consistent. We're not insisting the change.

**Comment by garydgregory** [comment]
Please rebase on master. Recent changes should allow all builds to be green including Java 16 and 17-EA.

**Comment by garydgregory** [comment]
I think the consensus is that is not really worth it. Closing.

**Pull request closed without merging**